src/main/java/com/redhat/devtools/intellij/qute/lang/format/QuteFormattingModelBuilder.java

```diff
@@ -0,0 +1,354 @@
+// Copyright 2000-2020 JetBrains s.r.o. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.redhat.devtools.intellij.qute.lang.format;
+
+import com.intellij.formatting.*;
+import com.intellij.formatting.templateLanguages.*;
+import com.intellij.lang.ASTNode;
+import com.intellij.openapi.util.TextRange;
+import com.intellij.psi.PsiFile;
+import com.intellij.psi.codeStyle.CodeStyleSettings;
+import com.intellij.psi.formatter.DocumentBasedFormattingModel;
+import com.intellij.psi.formatter.FormattingDocumentModelImpl;
+import com.intellij.psi.formatter.common.AbstractBlock;
+import com.intellij.psi.formatter.xml.HtmlPolicy;
+import com.intellij.psi.formatter.xml.SyntheticBlock;
+import com.intellij.psi.templateLanguages.SimpleTemplateLanguageFormattingModelBuilder;
+import com.intellij.psi.tree.IElementType;
+import com.intellij.psi.xml.XmlTag;
+import com.redhat.devtools.intellij.qute.lang.psi.QuteElementTypes;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.List;
+
+import static com.intellij.psi.formatter.WrappingUtil.getWrapType;
+
+/**
+ * Template aware formatter which provides formatting for Qute syntax and delegates formatting
+ * for the templated language to that languages formatter
+ *
+ * This class is a copy/paste from https://github.com/JetBrains/intellij-plugins/blob/master/Qute/src/com/dmarcotte/Qute/format/HbFormattingModelBuilder.java adapted for Qute.
+ */
+public class QuteFormattingModelBuilder extends TemplateLanguageFormattingModelBuilder {

```

Comment:
Should it be QuteHtmlFormattingModelBuilder? since some code handles XML tags

Reply:
I would like to keep like this, since Qute can be used with Yaml, Json, etc

Reply:
I rename to QuteHtmlFormattingModelBuilder and remove all unused code